setup.py

```diff
@@ -73,56 +75,74 @@ def _gen_grpc(self):
 
         proto_root_dir = root / 'azure_functions_worker' / 'protos'
         proto_src_dir = proto_root_dir / '_src' / 'src' / 'proto'
-        staging_root_dir = root / 'build' / 'protos'
+        build_dir = root / 'build'
+        staging_root_dir = build_dir / 'protos'
         staging_dir = (staging_root_dir
                        / 'azure_functions_worker' / 'protos')
-        build_dir = staging_dir / 'azure_functions_worker' / 'protos'
-        built_protos_dir = root / 'build' / 'built_protos'
-        built_proto_files_dir = (built_protos_dir
-                                 / 'azure_functions_worker' / 'protos')
+        built_protos_dir = build_dir / 'built_protos'
 
-        if os.path.exists(build_dir):
+        if os.path.exists(staging_root_dir):
             shutil.rmtree(build_dir)
 
         if os.path.exists(built_protos_dir):
             shutil.rmtree(built_protos_dir)
 
-        proto_files = glob.glob(str(proto_src_dir / '**' / '*.proto'),
-                                recursive=True)
+        shutil.copytree(proto_src_dir, staging_dir)
```

Comment:
Doesn't seems like **staging_dir** is used anywhere else. Maybe we can remove it?

Reply:
It's being used to do the copying as you pointed here. In the actual build command, we are using the same directory structure without explicitly referencing the `staging_dir`. Might be useful to walk through this offline as it's slightly tedious to explain here.